### PR TITLE
Fix assurance doubling on faint

### PR DESCRIPTION
**Pull request by whalemer**

### CODE REVIEW

**Comment by whalemer** [comment]
actually on second thoughts this doesn't work, i'm not sure how to fix it

File: data/moves.js [hunk 1]
```diff
@@ -677,20 +677,17 @@ let BattleMovedex = {
 		flags: {contact: 1, protect: 1, mirror: 1},
 		beforeTurnCallback: function (pokemon, target) {
 			pokemon.addVolatile('assurance');
-			pokemon.volatiles.assurance.position = target.position;
+			/** @type {boolean?[]} */
```
Comment (sirDonovan) [inline data/moves.js]: You can't use <code>@type</code> this way. It needs to go in the <code>Pokemon</code> constructor, but <code>volatiles</code> is not currently typed anyway.

File: package.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
   },
   "scripts": {
     "start": "node pokemon-showdown",
-    "test": "eslint --cache . && tsc && mocha",
```
Comment (sirDonovan) [inline package.json]: Also don't forget to revert this.

**Comment by Marty-D** [comment]
What sucks about Assurance is, like a few other things, its original implementation assumes Doubles doesn't exist. It also looks like it won't double at all when called by other moves.

This fixes the most glaring issue though, thanks!

**Pull request merged**